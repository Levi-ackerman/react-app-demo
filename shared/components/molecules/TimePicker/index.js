import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import RDatePicker from 'react-datepicker'
import 'react-datepicker/dist/react-datepicker.css'
import Icon from '../../atoms/Icon'
import moment from 'moment'

const Wrap = styled.div`
  border: 1px solid #e0e0e0;
  border-radius: 6px;
  height: 56px;
  padding: 0 16px;
  display: flex;
  align-items: center;
  background-color: #ffffff;
`

const Input = styled.input`
  color: #2d2d2d;
  font-size: 16px;
  margin-right: 16px;
  border: none;
  &:focus {
    outline: none;
  }
  ::placeholder {
    color: #939393;
    opacity: 1; /* Firefox */
  }

  :-ms-input-placeholder {
    /* Internet Explorer 10-11 */
    color: #939393;
  }

  ::-ms-input-placeholder {
    /* Microsoft Edge */
    color: #939393;
  }
`
const FixDatePickerTimer = styled.span`
  &
    .react-datepicker__time-container
    .react-datepicker__time
    .react-datepicker__time-box
    ul.react-datepicker__time-list
    li.react-datepicker__time-list-item {
    display: inline-block;
    margin-right: 0;
    color: #939393;
    font-size: 16px;
  }
  .react-datepicker__time-container {
    min-width: 100px;
  }
  .react-datepicker__time-container
    .react-datepicker__time
    .react-datepicker__time-box
    ul.react-datepicker__time-list {
    padding-right: 0;
  }
  .react-datepicker__time-container
    .react-datepicker__time
    .react-datepicker__time-box {
    width: auto;
  }
  .react-datepicker__time-container
    .react-datepicker__time
    .react-datepicker__time-box
    ul.react-datepicker__time-list
    li.react-datepicker__time-list-item:hover {
    background-color: #f7f7f7;
  }
`

class CustomInput extends React.PureComponent {
  render() {
    const { placeholder, value, onClick, onChange } = this.props
    return (
      <Wrap onClick={onClick}>
        <Input
          placeholder={placeholder}
          onChange={onChange}
          value={value}
          disabled
        />
        <Icon icon="down-arrow" />
      </Wrap>
    )
  }
}

class TimePicker extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      selected: props.selected ? props.selected : moment(),
    }
  }

  handleChange = date => {
    this.setState(
      {
        selected: date,
      },
      () => {
        if (this.props.onChange) {
          this.props.onChange(date)
        }
      },
    )
  }

  render() {
    return (
      <FixDatePickerTimer>
        <RDatePicker
          {...this.props}
          customInput={<CustomInput />}
          placeholderText="Select Time"
          showTimeSelect
          showTimeSelectOnly
          timeIntervals={30}
          dateFormat="LT"
          timeCaption="Time"
          selected={this.state.selected}
          onChange={this.handleChange}
        />
      </FixDatePickerTimer>
    )
  }
}

TimePicker.propTypes = {
  selected: PropTypes.object,
  onChange: PropTypes.func,
}

export default TimePicker
