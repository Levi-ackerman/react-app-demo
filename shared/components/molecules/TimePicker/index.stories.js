import React from 'react'
import { storiesOf } from '@storybook/react'
import moment from 'moment'
import TimePicker from '.'

storiesOf('TimePicker', module).add('default', () => (
  <TimePicker selected={moment()} />
))
