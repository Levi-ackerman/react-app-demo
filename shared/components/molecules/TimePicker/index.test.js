import React from 'react'
import renderer from 'react-test-renderer'
import TimePicker from '.'
import moment from 'moment'

describe('TimePicker', () => {
  it('renders correctly', () => {
    const date = moment('2018-05-01 00:00:00')
    const tree = renderer.create(<TimePicker selected={date} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
