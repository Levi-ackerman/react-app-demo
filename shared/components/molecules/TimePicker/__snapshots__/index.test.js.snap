// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimePicker renders correctly 1`] = `
<span
  className="TimePicker__FixDatePickerTimer-earsKW gkECMw"
>
  <div>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <div
          className="TimePicker__Wrap-dKmYR kHfeZW"
          onClick={[Function]}
        >
          <input
            className="TimePicker__Input-dpAhxb lblIJR"
            disabled={true}
            onChange={[Function]}
            placeholder="Select Time"
            value="12:00 AM"
          />
          <span
            className="Icon__Wrapper-bdEMVZ kRGjUp"
            dangerouslySetInnerHTML={
              Object {
                "__html": "/asset/mock",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</span>
`;
